web-common/src/features/dashboards/filters/dimension-filters/DimensionFilter.spec.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  addFilter,
+  useDashboardFetchMocksForComponentTests,
+} from "@rilldata/web-common/features/dashboards/filters/test/filter-test-utils";
+import { renderFilterComponent } from "@rilldata/web-common/features/dashboards/filters/test/render-filter-component";
+import {
+  createAndExpression,
+  createInExpression,
+  createLikeExpression,
+} from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import {
+  AD_BIDS_EXPLORE_INIT,
+  AD_BIDS_EXPLORE_NAME,
+  AD_BIDS_METRICS_INIT,
+  AD_BIDS_METRICS_NAME,
+  AD_BIDS_PUBLISHER_DIMENSION,
+} from "@rilldata/web-common/features/dashboards/stores/test-data/data";
+import { mockAnimationsForComponentTesting } from "@rilldata/web-common/lib/test/mock-animations";
+import { act, fireEvent, screen, waitFor } from "@testing-library/svelte";
+import { get } from "svelte/store";
+import { beforeAll, describe, expect, it } from "vitest";
+
+describe("DimensionFilter", () => {
+  mockAnimationsForComponentTesting();
+  const mocks = useDashboardFetchMocksForComponentTests();
+
+  beforeAll(() => {
+    mocks.mockMetricsView(AD_BIDS_METRICS_NAME, AD_BIDS_METRICS_INIT);
+    mocks.mockMetricsExplore(
+      AD_BIDS_EXPLORE_NAME,
+      AD_BIDS_METRICS_INIT,
+      AD_BIDS_EXPLORE_INIT,
+    );
+  });
+
+  it("Select mode filter", async () => {
```

Comment:
This is a bit hard to follow. At first, given the name of the test. I was confused when I saw "Contains" logic. It'd help if there were sectional comments at the _beginning_ of the "Contains" and "In list" parts of the test, not just at the end where those assertion happens.

Reply:
Sectional comments would help within each of this file's tests